### PR TITLE
Fix bugs in IOSPs' dataset rehydration

### PR DESCRIPTION
**Pull request by cwardgar**
Originally from #574 (commits applied to 5.0.0). I'm just cherry-picking them onto master.

### CODE REVIEW

File: cdm/src/main/java/ucar/nc2/iosp/nids/Nidsiosp.java [hunk 1]
```diff
@@ -1937,32 +1937,10 @@ int getInt(byte[] b, int offset, int num) {
...
-
-    //List alist = ncf.getGlobalAttributes();
-
-    ucar.nc2.Variable v = ncf.findVariable("BaseReflectivity");
-
-    int[] origin = {0, 0};
-    int[] shape = {300, 36};
-
-    v.read(origin, shape);
-    ncf.close();
```
Comment (dopplershift) [inline cdm/src/main/java/ucar/nc2/iosp/nids/Nidsiosp.java]: What, you didn't like the use main method test? 😈

Comment (lesserwhirls) [inline cdm/src/main/java/ucar/nc2/iosp/nids/Nidsiosp.java]: ryan

**Comment by lesserwhirls** [comment]
@cwardgar - let's do this one next.

**Comment by cwardgar** [comment]
Rebased.

**Pull request merged**